homeassistant/components/nextdns/sensor.py

```diff
@@ -0,0 +1,299 @@
+"""Support for the NextDNS service."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import cast
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import NextDnsUpdateCoordinator
+from .const import (
+    ATTR_DNSSEC,
+    ATTR_ENCRYPTION,
+    ATTR_IP_VERSIONS,
+    ATTR_PROTOCOLS,
+    ATTR_STATUS,
+    DOMAIN,
+)
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass
+class NextDnsSensorRequiredKeysMixin:
+    """Class for NextDNS entity required keys."""
+
+    coordinator_type: str
+
+
+@dataclass
+class NextDnsSensorEntityDescription(
+    SensorEntityDescription, NextDnsSensorRequiredKeysMixin
+):
+    """NextDNS sensor entity description."""
+
+
+SENSORS = (
+    NextDnsSensorEntityDescription(
+        key="all_queries",
+        coordinator_type=ATTR_STATUS,
+        entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
Are all of these really for diag?

Reply:
In my opinion, yes, this is information about the number of queries with different criteria. Do you think another entity category should be used?

Reply:
I don't know. Just asking since it appears a bit strange all sensors goes as diag and not any off them are without a category.